crates/authifier/src/database/definition.rs

```diff
@@ -19,6 +19,9 @@ pub trait AbstractDatabase: std::marker::Sync {
         normalised_email: &str,
     ) -> Result<Option<Account>>;
 
+    /// Find account by SSO ID
+    async fn find_account_by_sso_id(&self, idp_id: &str, sub_id: &str) -> Result<Option<Account>>;
```

Comment:
Every token/userinfo response should have a subject claim which can be utilized to identify the user, in this case we need to know the `idp_id` i.e. Google/GitHub/etc. and the `sub_id` (should be `serde_json::Value`).